idunn/utils/circuit_breaker.py

```diff
@@ -24,11 +22,42 @@ def state_change(self, cb, old_state, new_state):
 
 
 class IdunnCircuitBreaker(pybreaker.CircuitBreaker):
-    def __init__(self, name):
+    def __init__(self, name, fail_max, reset_timeout):
         super().__init__(
-            fail_max=settings["CIRCUIT_BREAKER_MAXFAIL"],
-            reset_timeout=settings["CIRCUIT_BREAKER_TIMEOUT"],
+            fail_max=fail_max,
+            reset_timeout=reset_timeout,
             exclude=[is_http_client_error],
             listeners=[LogListener()],
             name=name,
         )
+
+    async def call_async(self, f, *args, **kwargs):
+        """
+        Run the circuit breaker with native python async.
+        """
+
+        def raise_err_callback(err):
+            def f():
+                raise err
+
+            return f
+
+        # Check that the circuit breaker is open
+        state = self.state
+
+        if isinstance(state, pybreaker.CircuitOpenState):
+            # This is a hack: `before_call` takes a function parameter that
+            # should not be used, this way we can reproduce synchronous
+            # behavior before running the function by raising the exact same
+            # exception.
+            state.before_call(None)
```

Comment:
The existing implementation from `pybreaker` wraps calls to `call` and `call_async` with a lock. 
Is that not a problem that we don't use this lock here ?

Reply:
I added the lock to get a similar behavior, I don't think we use any multiprocessing so we may not have any concurrency issues anyway?